### ISSUE TITLE
Inline Notifications - Create Figma component and guidance

### ISSUE THREAD

**Issue by KennyAtHPE** [grommet/hpe-design-system#2886]
Ticket t-shirt sizing - Medium to Large 
Requires effort from both a designer and developer

Notification inline - draft single inline notification, Figma, design system site example, guidance

## Acceptance Criteria

- [x] Figma component for review
- [x] ~~design system site pull request with example and~~ guidance ready for internal review
- [x] follow on issues created as needed

## not in scope

- multiple inline notifications, either stacked or sequenced

------------------------------

## Guidance

**All terminology TBD**

Two main use cases/placements
- Page-Level
    - Underneath the page header
    - Summary of the whole page 
- Contextual-Level 
    - Typically above but can be below related content

Anatomy
- Icon/Status Indicator
    - Avoid using success/good status notifications, try Toast instead
- Message
    - General recommendation is to keep under 192 characters but it really depends on the placement. Sometimes it can be more, sometimes it should be less (do's and don'ts needed for this)
- Anchors (optional)
    - Use anchors to lead users to more details on the notification and/or to provide quick access to where the user can resolve an issue
- Close (optional)
    - Don't use for critical notifications or higher priority notifications
    - Documented within the notification center

When to use
- Higher priority and prominence notifications than Toast
    - Ask yourself, how important is this notification and whether or not you need it to be persistent?
        -  Important and should be persistent = Inline notifications
        - Not as important and doesn't matter if its persistent = Toast
- Notification that requires more text than Toast
- Used to show that a long process has been kicked off

When not to use
- Do not use it as a progress bar, should not be constantly updating itself in real time
- Do not use it as a post-submit indicator
    - Do's/Don'ts - show example of when we should use the button post submit indicator instead

Behavior
- Scale alongside content
    - If underneath the page header, it should fill the same width (What if the message is short?)
    - Do not put inside other components such as cards
    - Keep in mind responsiveness, inline banner notifications should not start out  extremely small
         - Connect with dev on possible solutions
    - Do not truncate text

Use Cases/Do's and Don'ts
- How to use with Tabs
- Task-generated and system-generated use cases
- How to use alongside other notifications
- Message content do's and don'ts

## Styling details

- Two variants?
    - More subtle
    - More prominence
    - Separate from form validation

- Border?
    - All around
    - Top, bottom, or left side only
 
- Background fill
    - Color based on icon/status level
    - Background/contrast
    - Background/back
    - No background
    
- Elevation?

- Rounding similar to Toast

**Comment by KennyAtHPE** [comment]
Moving this back into the backlog for now. Will prioritize this after #2889 is complete

**Comment by KennyAtHPE** [comment]
Dropping in some use cases shared by Cynthia to consider:

How should, if at all, inline notifications appear in context within cards? 



![Image](https://user-images.githubusercontent.com/86075038/216534660-e0a6e487-b339-45c0-be84-ffe5908ada1a.png)
Where should an inline notification be placed if there is an issue within the "Billing & vCPU Usage" card.



![Image](https://user-images.githubusercontent.com/86075038/216534884-54060dac-fe8b-4b49-94cd-def6662e27a7.png)
Where should an inline notification be placed if there is an issue within one of the three cards? What if there are multiple errors across multiple cards?

**Comment by KennyAtHPE** [comment]
**Mid-sprint status update**

Medium confidence that this will be completed by the end of the sprint. Planning to send out something to the team for review by the end of the week

Will break this ticket up according to how much work is left at the end of the sprint.

**Comment by KennyAtHPE** [comment]
This ticket will not be finalized by the sprint demo, aiming to have it ready for review by then instead.

**Comment by KennyAtHPE** [comment]
03/01 update

 Next steps after discussing with Scott:
- [ ] Review with the team by end of the week
- [ ] Create follow on ticket to connect w/ dev on creating a design system site pull request with examples and guidance

**Comment by KennyAtHPE** [comment]
03/03 Update:

Sent to the team for internal review. Some sections still need to be refined including:
- [ ] Top of page example
- [ ] Do's and Don'ts
- [ ] Variants
- [ ] Accessibility (will connect w/ Bill)

Will be creating follow on ticket with Mike once I'm back on Wednesday.

[WIP Guidance](https://www.figma.com/file/FGvOd2htjrChVVS2oz73ds/HPE-Notifications?node-id=5091%3A27871&t=udrlVtvxKTs8Rzri-1)
[Component](https://www.figma.com/file/FGvOd2htjrChVVS2oz73ds/HPE-Notifications?node-id=5091%3A29442&t=udrlVtvxKTs8Rzri-1)

**Comment by KennyAtHPE** [comment]
Pinging @SOjaHPE @ericsoderberghp @MikeKingdom for review on guidance.

Pinging@vavalos5 for review on the component, and if you have time, on the guidance as well.

**Comment by vavalos5** [comment]
Kenny, this is looking really great! The component works well. I left some comments on your Figma file. 😄

**Comment by ericsoderberghp** [comment]
Dropped some comments in Figma.

**Comment by KennyAtHPE** [comment]
3/9 update:

- Connected with Mike and Scott on this and reviewed comments from Eric and Vicky. Thanks for the help, everyone.
- @MikeKingdom will be helping me create the follow on implementation ticket.

**Comment by KennyAtHPE** [comment]
Next Steps
#3215, #3216 

Closing out ticket.
